Can we bump up UnCSS package version?

## Bump up UnCSS package version!

Having some problem in installing dependencies by `npm install` here

```md
[samar@sp gulp-uncss [±] master 6a1f22c ✓]$ npm install
/
> phantomjs@1.9.12 install /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/phantomjs
> node install.js

Writing location.js file
PhantomJS is already installed at /usr/local/bin/phantomjs.
npm ERR! Darwin 14.0.0
npm ERR! argv "node" "/Users/samar/github/nvm/v0.10.22/bin/npm" "install"
npm ERR! node v0.10.22
npm ERR! npm  v2.1.10
npm ERR! path /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander/package.json
npm ERR! code ENOENT
npm ERR! errno 34

npm ERR! enoent ENOENT, open '/Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander/package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/samar/github/gulp-uncss/npm-debug.log

```

### npm-debug.log with version `0.8.1`

```log
26314 info postinstall phantomjs@1.9.12
26315 verbose unlock done using /Users/samar/.npm/_locks/phantomjs-edb554be09331607.lock for /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/phantomjs
26316 verbose about to build /Users/samar/github/gulp-uncss/node_modules/uncss
26317 verbose unlock done using /Users/samar/.npm/_locks/uncss-1df5aa28cc56f68e.lock for /Users/samar/github/gulp-uncss/node_modules/uncss
26318 verbose stack Error: ENOENT, open '/Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander/package.json'
26319 verbose cwd /Users/samar/github/gulp-uncss
26320 error Darwin 14.0.0
26321 error argv "node" "/Users/samar/github/nvm/v0.10.22/bin/npm" "install"
26322 error node v0.10.22
26323 error npm  v2.1.10
26324 error path /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander/package.json
26325 error code ENOENT
26326 error errno 34
26327 error enoent ENOENT, open '/Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander/package.json'
26327 error enoent This is most likely not a problem with npm itself
26327 error enoent and is related to npm not being able to find a file.
26328 verbose exit [ 34, true ]
26329 verbose unbuild node_modules/uncss/node_modules/commander
26330 verbose gentlyRm vacuuming /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander
26331 silly gentlyRm purging /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/commander
26332 silly gentlyRm quitting because other entries in /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules
26333 verbose unbuild node_modules/uncss
26334 info preuninstall uncss@0.8.1
26335 info uninstall uncss@0.8.1
26336 verbose unbuild rmStuff uncss@0.8.1 from /Users/samar/github/gulp-uncss/node_modules
26337 verbose gentlyRm verifying that /Users/samar/github/gulp-uncss is managed by npm
26338 silly cacheAndTest /Users/samar/github/gulp-uncss is not inside /Users/samar/github/gulp-uncss/node_modules
26339 silly isManaged /Users/samar/github/gulp-uncss is not inside /Users/samar/github/gulp-uncss/node_modules
26340 silly cacheAndTest /Users/samar/github/gulp-uncss is not inside /Users/samar/github/gulp-uncss/node_modules/.bin
26341 silly cacheAndTest /Users/samar/github/gulp-uncss is inside /Users/samar/github/gulp-uncss
26342 silly gentlyRm /Users/samar/github/gulp-uncss is managed by npm
26343 silly gentlyRm /Users/samar/github/gulp-uncss/node_modules/.bin/uncss is under /Users/samar/github/gulp-uncss
26344 verbose gentlyRm vacuuming /Users/samar/github/gulp-uncss/node_modules/.bin/uncss up to /Users/samar/github/gulp-uncss
26345 info postuninstall uncss@0.8.1
26346 verbose gentlyRm vacuuming /Users/samar/github/gulp-uncss/node_modules/uncss
26347 silly gentlyRm purging /Users/samar/github/gulp-uncss/node_modules/uncss
26348 silly gentlyRm quitting because other entries in /Users/samar/github/gulp-uncss/node_modules
```

### Fix

Updating uncss version to `0.9.1` fixes this problem for me.

```log
[samar@sp gulp-uncss [±] master 6a1f22c ✓]$ git diff package.json
diff --git a/package.json b/package.json
index d40cd1e..6e4143d 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
   "dependencies": {
     "gulp-util": "~3.0.1",
     "object-assign": "^1.0.0",
-    "uncss": "0.8.1"
+    "uncss": "0.9.1"
   },
   "devDependencies": {
     "chai": "~1.9.1",

[samar@sp gulp-uncss [±] master 6a1f22c ✗]$ npm install
|
> phantomjs@1.9.12 install /Users/samar/github/gulp-uncss/node_modules/uncss/node_modules/phantomjs
> node install.js

Writing location.js file
PhantomJS is already installed at /usr/local/bin/phantomjs.
uncss@0.9.1 node_modules/uncss
├── commander@2.3.0
├── underscore@1.7.0
├── async@0.9.0
├── is-html@1.0.0 (html-tags@1.1.0)
├── node-phantom-simple@1.2.0
├── glob@4.3.1 (inherits@2.0.1, inflight@1.0.4, once@1.3.1, minimatch@2.0.1)
├── css@2.1.0 (inherits@2.0.1, urix@0.1.0, source-map-resolve@0.3.1, source-map@0.1.40)
├── request@2.44.0 (caseless@0.6.0, json-stringify-safe@5.0.0, forever-agent@0.5.2, aws-sign2@0.5.0, stringstream@0.0.4, oauth-sign@0.4.0, tunnel-agent@0.4.0, node-uuid@1.4.1, qs@1.2.2, mime-types@1.0.2, form-data@0.1.4, bl@0.9.3, http-signature@0.10.0, tough-cookie@0.12.1, hawk@1.1.1)
└── phantomjs@1.9.12 (which@1.0.7, progress@1.1.8, rimraf@2.2.8, ncp@1.0.1, kew@0.4.0, request-progress@0.3.1, mkdirp@0.5.0, adm-zip@0.4.4, npmconf@2.0.9, request@2.42.0)
```

--- a/README.md
+++ b/README.md
@@ -12,56 +12,47 @@ npm install gulp-uncss --save-dev
 
 ## Example
 
+Single files, globbing patterns and URLs are all supported by gulp-uncss, and can be mixed and matched:
+
 ```js
 var gulp = require('gulp');
 var uncss = require('gulp-uncss');
 
-gulp.task('simple', function() {
+gulp.task('default', function() {
     return gulp.src('site.css')
         .pipe(uncss({
-            html: ['index.html', 'about.html']
+            html: ['index.html', 'posts/**/*.html', 'http://example.com']
         }))
         .pipe(gulp.dest('./out'));
 });
 ```
 
-## Glob example
-
-UnCSS does not provide native support for globbing patterns. If you would like gulp-uncss to parse a directory recursively, then you can use the [glob module](https://www.npmjs.org/package/glob) like so:
+gulp-uncss can also be used in a pipeline that involves CSS pre-processing. Utilising many transforms on a single file is one of gulp's strengths:
 
 ```js
-var glob = require('glob');
+var gulp = require('gulp');
+var uncss = require('gulp-uncss');
+var sass = require('gulp-sass');
+var concat = require('gulp-concat');
+var csso = require('gulp-csso');
 
-gulp.task('glob', function() {
-    gulp.src('site.css')
+gulp.task('default', function() {
+    return gulp.src('styles/**/*.scss')
+        .pipe(sass())
+        .pipe(concat('main.css'))
         .pipe(uncss({
-            html: glob.sync('templates/**/*.html')
+            html: ['index.html', 'posts/**/*.html', 'http://example.com']
         }))
+        .pipe(csso())
         .pipe(gulp.dest('./out'));
 });
 ```
 
-## URL example
-
-UnCSS can also visit your website for the HTML it uses to analyse the CSS against. Here is an example:
-
-```js
-gulp.task('urls', function() {
-    gulp.src('site.css')
-        .pipe(uncss({
-            html: [
-                'http://www.example.com'
-            ]
-        }))
-        .pipe(gulp.dest('./out'));
-});
-```
-
-Note that you can mix and match URLs and paths to files using the `html` option.
+In just a few lines, we compiled SCSS source into a single file, removed unused CSS and minified the output!
 
 ## Options
 
-This plugin takes slightly different options to the UnCSS module, because it is essentially just a streaming wrapper which returns a CSS stream.
+Please see the [UnCSS documentation](https://github.com/giakki/uncss#within-nodejs) for all of the options you can use. Some of them aren't as necessary when using gulp-uncss, because the CSS to analyse comes from the stream rather than the HTML files. The main options you will likely be using are:
 
 ### html
 Type: `Array|String`
@@ -80,6 +71,8 @@ Type: `Integer`
 Default value: `undefined`
 
 Specify how long to wait for the JS to be loaded.
+
+Note that `options.ignoreSheets` is *already defined* for you. gulp-uncss will only process CSS files in the stream.
 
 ## Contributing
 

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -2,24 +2,20 @@
 
 'use strict';
 
-var gulp    = require('gulp'),
-    gutil   = require('gulp-util'),
-    clear   = require('clear'),
-    mocha   = require('gulp-mocha'),
-    jshint  = require('gulp-jshint');
+var gulp   = require('gulp'),
+    mocha  = require('gulp-mocha'),
+    jshint = require('gulp-jshint');
 
-gulp.task('lint', function () {
-    gulp.src('*.js')
-        .pipe(jshint())
-        .pipe(jshint.reporter('jshint-stylish'))
-        .pipe(mocha());
-});
+gulp.task('develop', function() {
+    function scripts() {
+        return gulp.src('*.js')
+            .pipe(jshint())
+            .pipe(jshint.reporter('jshint-stylish'))
+            .pipe(mocha({ reporter: 'mocha-silent-reporter' }));
+    }
 
-gulp.task('default', function() {
-    gulp.run('lint');
-    gulp.watch('*.js', function(event) {
-        clear();
-        gutil.log(gutil.colors.cyan(event.path.replace(process.cwd(), '')) + ' ' + event.type + '. (' + gutil.colors.magenta(gutil.date('HH:MM:ss')) + ')');
-        gulp.run('lint');
-    });
+    var watcher = gulp.watch('*.js');
+    watcher.on('change', scripts);
+
+    return scripts();
 });

--- a/index.js
+++ b/index.js
@@ -2,27 +2,24 @@
 
 'use strict';
 
-var uncss           = require('uncss'),
-    gutil           = require('gulp-util'),
-    transform       = require('stream').Transform,
-    objectAssign    = require('object-assign'),
+var uncss       = require('uncss'),
+    gutil       = require('gulp-util'),
+    assign      = require('object-assign'),
+    transform   = require('stream').Transform,
 
-    PLUGIN_NAME     = 'gulp-uncss';
+    PLUGIN_NAME = 'gulp-uncss';
 
-module.exports = function() {
+module.exports = function(options) {
     var stream = new transform({ objectMode: true });
-    var options = {
-        html: arguments[0].html,
-        ignore: arguments[0].ignore,
-        timeout: arguments[0].timeout,
-        ignoreSheets: [/\s*/]
-    };
+
+    // Ignore stylesheets in the HTML files; only use those from the stream
+    options.ignoreSheets = [/\s*/];
 
     stream._transform = function(file, unused, done) {
         if (file.isStream()) {
             return done(new gutil.PluginError(PLUGIN_NAME, 'Streaming not supported'));
         } else if (file.isBuffer()) {
-            uncss(options.html, objectAssign(options, { raw: String(file.contents) }), function(err, output) {
+            uncss(options.html, assign(options, { raw: String(file.contents) }), function(err, output) {
                 if (err) {
                     return done(new gutil.PluginError(PLUGIN_NAME, err));
                 }
@@ -36,4 +33,4 @@ module.exports = function() {
     };
 
     return stream;
-}
+};

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
   "dependencies": {
     "gulp-util": "~3.0.1",
     "object-assign": "^2.0.0",
-    "uncss": "^0.11.0"
+    "uncss": "^0.12.0"
   },
   "devDependencies": {
     "chai": "~1.10.0",

--- a/package.json
+++ b/package.json
@@ -28,12 +28,13 @@
   },
   "devDependencies": {
     "chai": "~1.10.0",
+    "event-stream": "~3.2.1",
     "gulp": "~3.8.10",
-    "gulp-mocha": "~2.0.0",
     "gulp-jshint": "~1.9.0",
+    "gulp-mocha": "~2.0.0",
     "jshint-stylish": "~1.0.0",
     "mocha": "~2.1.0",
-    "event-stream": "~3.2.1"
+    "mocha-silent-reporter": "^1.0.0"
   },
   "main": "index.js"
 }

--- a/package.json
+++ b/package.json
@@ -24,17 +24,16 @@
   "dependencies": {
     "gulp-util": "~3.0.1",
     "object-assign": "^2.0.0",
-    "uncss": "0.8.1"
+    "uncss": "^0.11.0"
   },
   "devDependencies": {
-    "chai": "~1.9.1",
-    "clear": "0.0.1",
-    "gulp": "~3.5.6",
-    "gulp-mocha": "~0.4.1",
-    "gulp-jshint": "~1.5.0",
-    "jshint-stylish": "~0.1.5",
-    "mocha": "~1.18.2",
-    "event-stream": "~3.1.0"
+    "chai": "~1.10.0",
+    "gulp": "~3.8.10",
+    "gulp-mocha": "~2.0.0",
+    "gulp-jshint": "~1.9.0",
+    "jshint-stylish": "~1.0.0",
+    "mocha": "~2.1.0",
+    "event-stream": "~3.2.1"
   },
   "main": "index.js"
 }

--- a/test.js
+++ b/test.js
@@ -5,7 +5,7 @@
 
 var expect = require('chai').expect,
     gutil  = require('gulp-util'),
-    uncss  = require('./index'),
+    uncss  = require('./'),
     Stream = require('stream'),
     es     = require('event-stream'),
 

